bin/test-workflow/7_app_deploy.sh

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+. utils.sh
+
+main() {
+  announce "Deploying test apps for $TEST_APP_NAMESPACE_NAME."
+
+  URLENCODED_AUTHN_ID=$(urlencode $AUTHENTICATOR_ID)
+
+  set_namespace $TEST_APP_NAMESPACE_NAME
+  init_registry_creds
+  init_connection_specs
+
+  if is_minienv; then
+    IMAGE_PULL_POLICY='Never'
+  else
+    IMAGE_PULL_POLICY='Always'
+  fi
+
+  deploy_app_backend
+  # deploy_secretless_app
+  deploy_sidecar_app
```

Comment:
I think we need a little more logic here.

Ultimately, the choice of which of the various flavors of app integrations should be deployed should be settable by whomever is calling these scripts (i.e. it's not reusable if we're relying on commenting/uncommenting the script here). I think the best way to do this would be to add environment variables to enable each of the app integrations (and then we can control it via Summon in the Jenkins CI).

So then the question is how do we use those new environment variable settings here in the script? One way we can do this, which matches what you're doing here, me thinks, is to conditionally and separately call individual bash deploy functions based on those environment variables, and each of those bash functions would be doing separate `helm install ...` invocations. So for example, if `DEPLOY_SIDECAR_APP` env variable is `true`, then we call `deploy_sidecar_app` function, and that does a `helm install ... ` with `--set app-summon-sidecar.enabled=true` and all other integrations disabled.

This would work, but the automated testing would then be installing N different Helm releases. Not a big deal. But if we do go this route, maybe we can rethink and simplify the way we're using a main app deploy Helm chart and multiple subcharts, since we'd really be using each subchart independently (subcharts aren't buying us anything?).

Another approach would be to check each of the new environment variables, and compile a set of `--set` arguments that might add up to e.g.:
```
--set "app-summon-sidecar.enable=true" \
--set "app-summon-init.enable=false" \
--set "app-secretless-broker.enable=true" \
--set "app-secrets-provider.enable=true"
```
(along with some other app/integration specific settings), and then we would do a single `helm install ...` on the main app-deploy Helm chart.

I'm not sure what the best approach is, I'm leaning towards a single `helm install ...`, but could be persuaded otherwise.

Reply:
An advantage to using a single `helm install ...` on the parent chart is that all subcharts would be installed in parallel. Another advantage is that we'd be able to create Kubernetes resources that are shared by all of the subcharts (e.g. one common TLS Secret for all backends).

Reply:
Added the `enabled` flag for `app-summon-sidecar` on the parent chart